x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -161,7 +162,7 @@ public final PhysicalOperation sourcePhysicalOperation(EsQueryExec esQueryExec,
         List<Sort> sorts = esQueryExec.sorts();
         assert esQueryExec.estimatedRowSize() != null : "estimated row size not initialized";
         int rowEstimatedSize = esQueryExec.estimatedRowSize();
-        int limit = esQueryExec.limit() != null ? (Integer) esQueryExec.limit().fold() : NO_LIMIT;
+        int limit = esQueryExec.limit() != null ? (Integer) esQueryExec.limit().fold(context.foldCtx()) : NO_LIMIT;
```

Comment:
The folds in this file also look like they're purely out of defensiveness.

Reply:
Right. I wanted to keep this change mechanical rather than take this all apart.

Reply:
++